### DISCUSSION TITLE
Background colors and header/sidenav

### DISCUSSION THREAD

**Discussion by mayank99** [iTwin/iTwinUI#103] [General]
_Continuation of https://github.com/iTwin/iTwinUI/pull/33#discussion_r625873527_

### Preface

Now that we've released header and sidenav components, this is important to bring up.

When we first started iTwinUI, this is what we had in mind:

![image](https://user-images.githubusercontent.com/9084735/119021586-978b6180-b96d-11eb-9bd4-dc12373f9f63.png)

- **body would use background-2**
- page header and table would use background-1
- active tabs (e.g. in sidenav) would use background-1, inactive tabs would use background-3
- empty space in sidenav would use background-4

<details>
<summary>variables defined in theme.scss</summary>

https://github.com/iTwin/iTwinUI/blob/a9fc7b64af0c0af7c202c37455c67e2babad72ce/src/style/theme.scss#L123-L127
</details>

### The problem 

For some reason, we are using background-1 in `.iui-body`, not background-2.

https://github.com/iTwin/iTwinUI/blob/a9fc7b64af0c0af7c202c37455c67e2babad72ce/src/style/global.scss#L5-L7

Currently, when header and sidenav are placed above background-1, they just look bad/wrong and there isn't enough contrast. So the user would have to know all this context and need to manually override with background-2 (and that's asking too much from the user).

### Solutions?

The "easy" solution would be to just change the default in `.iui-body` to background-2. Since we are pre-1.0, this wouldn't be a big deal for the css package. But iTwinUI-react is at 1.X and is advertised as "stable", so doing this would be a huge visual change, affecting every user. If we do this, we'd probably have to make a bigger announcement.

~~However, header and sidenav were just released and don't have many users, so we can change the styles there. Specifically, we can explore using something that looks good against a body with background-1.~~

Something else to also keep in mind when making this decision is how it could affect high-contrast themes (when we add those in the future).

#### Update

_See #163_ 😄

**Comment by FlyersPh9** [comment]
We setup the demo pages to use background-1 because, while most components can be placed above background-1 or background-2, they most often will be placed above background-1.  However, when building a new iTwin application the HTML's `<body>` should be using background-2.  Upon background-2 you can place some titles, some navigation buttons, but most user interactions should take place within tables, tiles, or blocks of content… all of which would use background-1.

I think it makes the most sense to change the body to background-2 to avoid confusion when building iTwin apps using iTwinUI.
